packages/nimble-components/package.json

```diff
@@ -59,6 +59,7 @@
     "@types/d3-scale": "^4.0.2",
     "@types/d3-selection": "^3.0.0",
     "@types/d3-zoom": "^3.0.0",
+    "d3-array": "^3.2.2",
```

Comment:
I was surprised that you did not need to import type definitions for this as well, i.e. `@types/d3-array`.

It looks like at some point '@types/d3' was added as a dependency. The goal of explicitly listing each type package and supported versions was to have reproducible builds / type definitions.

The `@types/d3` package however pulls in any version of the other type definitions as you can see in the [package.json](https://unpkg.com/browse/@types/d3@7.4.0/package.json), i.e.:
```
        "@types/d3-array": "*",
```

So we aren't getting a reproducible set of type definition versions, we just get whatever happens to be the latest.

Can you remove `@types/d3` and add:
```
    "@types/d3-array": "^3.0.4",
    "@types/d3-random": "^3.0.1",
```